starlite/routes/base.py

```diff
@@ -98,38 +102,51 @@ def create_handler_kwargs_model(self, route_handler: "BaseRouteHandler") -> Kwar
-                raise ImproperlyConfiguredException("Path parameter names should be of length greater than zero")
-            if param_type not in param_type_map:
-                raise ImproperlyConfiguredException(
-                    "Path parameters should be declared with an allowed type, i.e. 'str', 'int', 'float' or 'uuid'"
-                )
+        if len(param.split(":")) != 2:
+            raise ImproperlyConfiguredException(
+                "Path parameters should be declared with a type using the following pattern: '{parameter_name:type}', e.g. '/my-path/{my_param:int}'"
+            )
+        param_name, param_type = (p.strip() for p in param.split(":"))
+        if len(param_name) == 0:
+            raise ImproperlyConfiguredException("Path parameter names should be of length greater than zero")
+        if param_type not in param_type_map:
+            raise ImproperlyConfiguredException(
+                "Path parameters should be declared with an allowed type, i.e. 'str', 'int', 'float' or 'uuid'"
+            )
 
     @classmethod
-    def _parse_path(cls, path: str) -> Tuple[str, str, List[PathParameterDefinition]]:
+    def _parse_path(cls, path: str) -> Tuple[str, str, List[Union[str, PathParameterDefinition]]]:
         """Normalizes and parses a path."""
         path = normalize_path(path)
-        path_format = path
-        path_parameters = []
-        identified_params = param_match_regex.findall(path)
-        cls._validate_path_parameters(identified_params)
-        for param in identified_params:
-            param_name, param_type = (p.strip() for p in param.split(":"))
-            path_format = path_format.replace(param, param_name)
-            path_parameters.append(
-                PathParameterDefinition(name=param_name, type=param_type_map[param_type], full=param)
-            )
-        return path, path_format, path_parameters
+
+        parsed_components: List[Union[str, PathParameterDefinition]] = []
+
+        # Handle each component in the route path
+        components = [component for component in path.split("/") if component]
+        for component in components:
+            param_match = param_match_regex.fullmatch(component)
+            if param_match:
+                # Parse and validate components that are path params
+                param = param_match.group(1)
+                cls._validate_path_parameter(param)
+                param_name, param_type = (p.strip() for p in param.split(":"))
+                parsed_components.append(
+                    PathParameterDefinition(name=param_name, type=param_type_map[param_type], full=param)
+                )
+            else:
+                # Just append strings
+                parsed_components.append(component)
+
+        # Build the url used for the OpenAPI schema
+        path_format = join_paths(
+            f"{{{component['name']}}}" if isinstance(component, dict) else component for component in parsed_components
```

Comment:
Looks somewhat weird TBH. Might be simpler to parse mentally if split into two assignments.